packages/backend/src/helpers/actions.ts

```diff
@@ -44,7 +44,12 @@ const CONNECTIVITY_ERROR_SIGNS = ['ETIMEDOUT', 'ECONNRESET']
 const CONNECTIVITY_STATUS_CODE = [504]
 
 export function handleErrorAndThrow(errorDetails: IJSONObject): never {
-  const errorString = get(errorDetails, 'details.error', '') as string
+  const errorVariable = get(errorDetails, 'details.error', '') as unknown
+  const errorString =
+    typeof errorVariable === 'string'
+      ? errorVariable
+      : JSON.stringify(errorVariable)
```

Comment:
nit: we die if errorVariable contains circular reference but unlikely that will happen. can surround with a try/catch just in case too if you are worried.